engine/src/main/java/io/camunda/zeebe/streamprocessor/BufferedStreamWriter.java

```diff
@@ -82,27 +84,24 @@ void appendRecord(
     writeMetadata(metadataLength);
     writeValue(valueWriter, valueLength);
 
-    eventLength += metadataLength + valueLength;
-    eventCount += 1;
+    recordCount += 1;
   }
 
   boolean canWriteAdditionalEvent(final int length) {
-    final var count = eventCount + 1;
-    final var batchLength = eventLength + length + (count * HEADER_BLOCK_LENGTH);
-    return batchLength < maxFragmentSize;
+    return length < capacityCalculator.apply(recordBufferOffset);
```

Comment:
:x: I expect the current function to map down to `Dispatcher#canClaimFragmentBatch(int recordCount, int batchLength)` at the end, so it should be a `BiFunction<Integer, Integer, Integer>`, I think.